ui/src/pages/organizations/details/edit/billing.tsx

```diff
@@ -72,74 +88,105 @@ export function EditBillingPanel({ onClose }: EditBillingPanelProps) {
+    {
+      enabled: !!organizationId && !!billingId,
+      select: (data) => {
+        const isPostpaid = data?.creditMin < 0n;
+        const creditMin = isPostpaid ? data.creditMin * BigInt(-1) : data.creditMin;
+        return {
+          tokenPaymentType: isPostpaid ? "postpaid" as const : "prepaid" as const,
+          creditMin: creditMin.toString(),
+          dueInDays: data?.dueInDays.toString(),
+        };
+      },
     },
-    [reset],
   );
 
   useEffect(() => {
-    if (billingAccount?.org_id && billingAccount?.id) {
-      getBillingDetails(billingAccount?.org_id, billingAccount?.id);
+    if (error) {
+      toast.error("Something went wrong", {
+        description: "Unable to fetch billing details",
+      });
+      console.error("Unable to fetch billing details:", error);
     }
-  }, [billingAccount?.org_id, billingAccount?.id, getBillingDetails]);
+  }, [error]);
 
-  const onSubmit = async (data: BillingDetailsForm) => {
-    // Transform data for API submission
-    const apiData = billingDetailsApiSchema.parse(data);
-
-    await api?.adminServiceUpdateBillingAccountDetails(
-      billingAccount?.org_id || "",
-      billingAccount?.id || "",
-      {
-        credit_min: apiData.credit_min.toString(),
-        due_in_days: apiData.due_in_days.toString(),
+  useEffect(() => {
+    if (billingDetailsData) {
+      reset(billingDetailsData);
+    }
+  }, [billingDetailsData, reset]);
+
+  const { mutateAsync: updateBillingDetails } = useMutation(
+    AdminServiceQueries.updateBillingAccountDetails,
+    {
+      onError: (error) => {
+        toast.error("Something went wrong", {
+          description: error.message,
+        });
+        console.error("Unable to update billing details:", error);
       },
-    );
-    const getBillingResp = await api?.frontierServiceGetBillingAccount(
-      billingAccount?.org_id || "",
-      billingAccount?.id || "",
-      { with_billing_details: true },
-    );
-    const updatedDetails = getBillingResp?.data?.billing_details;
-    if (updatedDetails && setBillingAccountDetails) {
-      setBillingAccountDetails(updatedDetails);
+    },
+  );
+
+  const onSubmit = async (data: BillingDetailsForm) => {
+    try {
+      // For prepaid, set values to 0; for postpaid, use form values
+      const creditMinValue = data.tokenPaymentType === "prepaid" ? 0n : BigInt(data.creditMin);
+      const dueInDaysValue = data.tokenPaymentType === "prepaid" ? 0n : BigInt(data.dueInDays);
+
+      // For postpaid, creditMin should be negative for API
+      const creditMinForApi = data.tokenPaymentType === "postpaid"
+        ? -creditMinValue
+        : creditMinValue;
+
+      await updateBillingDetails(
+        create(UpdateBillingAccountDetailsRequestSchema, {
+          orgId: organizationId,
+          id: billingId,
+          creditMin: creditMinForApi,
+          dueInDays: dueInDaysValue,
+        }),
+      );
+
+      const getBillingDetailsResp = await refetchBillingDetails();
```

Comment:
why are we refetching billing details. can we use the response of `updateBillingDetails` mutaion here?